### ISSUE TITLE
fn assignment

### ISSUE THREAD

**Issue by Tarang11** [vlang/v#5715]
**V version:** Latest
**OS:** Mac OS

**What did you do?**

fn test() int {
	return 113
}

fn test1() int {
	return 1113
}

fn main() {
	mut a := 0
	mut b := 0
	a, b = test(), test1()
	println("$a, $b")
}

and 

fn main() {
	mut d := [10, 12]
	d[0], d[1] = test(), test1()
	println(d)
}

**What did you expect to see?**
Return value assigned to a, b.
 
**What did you see instead?**
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

If you were not working with C interop, please raise an issue on GitHub:

https://github.com/vlang/v/issues/new/choose`

**Comment by bashery** [comment]
The same problem in Linux.

```
a, b = init(), 321  // works.
a, b = 321, init() //  works,
a,b = init(), init() // error``
```

**Comment by yuyi98** [comment]
Fixed by #5716
